packages/vite/src/node/build.ts

```diff
@@ -390,6 +401,10 @@ async function doBuild(
     )
   }
 
+  if (options.optimizeDeps && !ssr) {
+    await createOptimizedDeps(config)
+  }
```

Comment:
`initOptimizedDeps` and `DepsOptimizer` would be great 😄  I also kind of feel like the optimizer should be a class instead, since it has a lot of moving parts between nested functions.

Reply:
Done in [refactor: optimizedDeps -> depsOptimizer](https://github.com/vitejs/vite/pull/8280/commits/e7cac66b445d5336890506035953c909ebe9a601), also finally renamed `registerMissingImport.ts` to `optimizer.ts`

Reply:
We could check in another PR if making it a class will end up making it easier to work with. I kind of like a plain object. But we can do a clean-up, it has been evolving a lot, and sometimes I tried to avoid bigger changes for the PRs to be easier to digest.